packages/core/src/node-execution-context/base-contexts.ts

```diff
@@ -0,0 +1,107 @@
+import type {
+	FunctionsBase,
+	INode,
+	INodeExecutionData,
+	IWorkflowExecuteAdditionalData,
+	NodeTypeAndVersion,
+	Workflow,
+	WorkflowExecuteMode,
+} from 'n8n-workflow';
+import { deepCopy, LoggerProxy } from 'n8n-workflow';
+import { Container } from 'typedi';
+
+import { InstanceSettings } from '@/InstanceSettings';
+
+export abstract class BaseContext implements Omit<FunctionsBase, 'getCredentials'> {
```

Comment:
Could we have documentation for this class

Reply:
I want to document all the context objects together in a single PR, once I understand this code better. Created a [ticket](https://linear.app/n8n/issue/CAT-280) for that.